testflinger_agent/__init__.py

```diff
@@ -90,18 +90,17 @@ def emit(self, record):
 
     def flush(self):
         """Flush and post buffer"""
-        try:
-            # atomic queue iteration
-            for record in list(self.reqbuffer):
+        for record in list(self.reqbuffer):
```

Comment:
Ohhh, ok that explains my confusion. I didn't catch this at first, but I think you are actually sending a new post for every single line in the log? we definitely shouldn't be doing that. The whole thing should be sent at once as a list of lines - whether it's 1 line or 50. This is why I was confused why you would want to remove them one line at a time. You will want to send the entire buffer, then definitely clear the whole thing - not send or remove them one line at a time.

Reply:
@hum4n0id Are you able to take a look at the changes I requested on this? I'd really like to get this landed soon. Thanks!